pkg/rpc/connection_class.go

```diff
@@ -45,6 +47,35 @@ const (
 	NumConnectionClasses int = iota
 )
 
+// classOverrides contains a mapping from ConnectionClass to ConnectionClass.
+var classOverrides = parseClassOverrides(
+	envutil.EnvOrDefaultString("COCKROACH_USE_DEFAULT_CONNECTION_CLASS", ""))
+
+// parseClassOverride parses the list of connection class overrides. See
+// classOverrides variable comment for syntax.
+func parseClassOverrides(str string) map[ConnectionClass]ConnectionClass {
+	overrides := make(map[ConnectionClass]ConnectionClass)
+	for str += ","; len(str) != 0; {
+		pos := strings.IndexByte(str, ',') // always finds a position
+		name := str[:pos]
+		if class, ok := connectionClassFromName[name]; ok {
+			overrides[class] = DefaultClass
+		}
+		// NB: unknown classes are skipped for extensibility.
+		str = str[pos+1:]
+	}
+	return overrides
+}
+
+// ConnectionClassOverride returns the RPC connection class that the passed in
+// class overrides to.
+func ConnectionClassOverride(c ConnectionClass) ConnectionClass {
+	if override, ok := classOverrides[c]; ok {
```

Comment:
Made it a map in this quick prototype, but we should really have `[]ConnectionClass` slice indexed by `int(ConnectionClass)`, because there are only a few connection classes with dense values.

Same applies to the `connectionClassName` map below. Better to have the lookup cost be negligible.